data/summary/store/store.go

```diff
@@ -80,6 +82,144 @@ func (r *TypelessRepo) DeleteSummary(ctx context.Context, userId string) error {
 	return nil
 }
 
+// isRealtimeUser processes two slices of Item and returns an int count of days with realtime records.
+// this currently doesn't handle N slices, only 1-2, might need adjustment for more types.
+func isRealtimeUser(a, b []*types.Bucket[*types.CGMBucketData, types.CGMBucketData]) int {
```

Comment:
The name and the signature are off. If a function starts with `is` it should return a boolean. Consider renaming this to `GetNumberOfDaysWithRealtimeData`.

Reply:
It is also not clear why you need to pass two slices.

Reply:
yeah, this is an artifact of me not reading the expected return json properly and thinking I only needed a boolean, i changed the function without changing the name, ill make it match.

as for the two slices, this is an optimization/simplification for now, as either bgm or cgm buckets can trigger the day marking, so doing one slice at a time would require returning a list of days, then de-duplicating. As a single pass, we can skip the rest of the buckets for the day for both types together, simplifying it to a counter.

In the future, which is scheduled for the Insulin Stats work, this will be moved to the DB with a single map reduce query or similar, as they will be in a single collection, making this much more straight forward.